Local build failure compiling for STM32 and Rasberry Pi Pico on Ubuntu 24.04

Building for STM32 on Ubuntu-24.04 with the distro package of `gcc-arm-none-eabi` (version 13.2.1) results in a build failure

```
...
[ 80%] Building C object src/libAtomVM/CMakeFiles/libAtomVM.dir/valueshashtable.c.obj
/home/builder/work/AtomVM/src/libAtomVM/term.c: In function 'term_funprint':
/home/builder/work/AtomVM/src/libAtomVM/term.c:385:46: error: expected ')' before 'PRIu64'
  385 |         return fun->print(fun, "#Ref<0.0.0.%" PRIu64 ">", ref_ticks);
      |                          ~                   ^~~~~~~
      |                                              )
/home/builder/work/AtomVM/src/libAtomVM/term.c:35:1: note: 'PRIu64' is defined in header '<inttypes.h>'; did you forget to '#include <inttypes.h>'?
   34 | #include <stddef.h>
  +++ |+#include <inttypes.h>
   35 | #include <stdio.h>
/home/builder/work/AtomVM/src/libAtomVM/term.c:385:44: warning: spurious trailing '%' in format [-Wformat=]
  385 |         return fun->print(fun, "#Ref<0.0.0.%" PRIu64 ">", ref_ticks);
      |                                            ^
/home/builder/work/AtomVM/src/libAtomVM/term.c:382:18: warning: unused variable 'ref_ticks' [-Wunused-variable]
  382 |         uint64_t ref_ticks = term_to_ref_ticks(t);
      |                  ^~~~~~~~~
make[2]: *** [src/libAtomVM/CMakeFiles/libAtomVM.dir/build.make:440: src/libAtomVM/CMakeFiles/libAtomVM.dir/term.c.obj] Error 1
make[1]: *** [CMakeFiles/Makefile2:218: src/libAtomVM/CMakeFiles/libAtomVM.dir/all] Error 2
make: *** [Makefile:91: all] Error 2

```

This similarly affects Raspberry Pi Pico builds:
```
...
/home/builder/work/AtomVM/src/libAtomVM/term.c: In function 'term_funprint':
/home/builder/work/AtomVM/src/libAtomVM/term.c:384:46: error: expected ')' before 'PRIu64'
  384 |         return fun->print(fun, "#Ref<0.0.0.%" PRIu64 ">", ref_ticks);
      |                          ~                   ^~~~~~~
      |                                              )
/home/builder/work/AtomVM/src/libAtomVM/term.c:34:1: note: 'PRIu64' is defined in header '<inttypes.h>'; did you forget to '#include <inttypes.h>'?
   33 | #include <stddef.h>
  +++ |+#include <inttypes.h>
   34 | #include <stdio.h>
/home/builder/work/AtomVM/src/libAtomVM/term.c:384:44: warning: spurious trailing '%' in format [-Wformat=]
  384 |         return fun->print(fun, "#Ref<0.0.0.%" PRIu64 ">", ref_ticks);
      |                                            ^
/home/builder/work/AtomVM/src/libAtomVM/term.c:381:18: warning: unused variable 'ref_ticks' [-Wunused-variable]
  381 |         uint64_t ref_ticks = term_to_ref_ticks(t);
      |                  ^~~~~~~~~

```

This seems to be a well known issue with recent `gcc-arm-none-eabi` and `newlib` releases for Debian as well as Ubuntu that affects any ARM platform builds, including Raspberry Pi Pico:
- [Debian "trixie" (testing)](https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1067692)
- Ubuntu-24.04 [workaround in pico-sdk](https://github.com/raspberrypi/pico-sdk/pull/1863) "develop" branch but still unreleased.

I tried using the work around implemented by the pico-sdk, including `sys/types.h` before `inttypes.h`, but this had no effect, and the error was still present when building for both platforms.

Example:
```
...
#include <ctype.h>
#include <sys/types.h>
#include <inttypes.h>
#include <stdarg.h>
...
```

--- a/.github/workflows/pico-build.yaml
+++ b/.github/workflows/pico-build.yaml
@@ -33,7 +33,7 @@ concurrency:
 
 jobs:
   pico:
-    runs-on: ubuntu-22.04
+    runs-on: ubuntu-24.04
     strategy:
       matrix:
         board: ["pico", "pico_w"]

--- a/.github/workflows/stm32-build.yaml
+++ b/.github/workflows/stm32-build.yaml
@@ -44,17 +44,6 @@ jobs:
           https://repo.hex.pm
           https://cdn.jsdelivr.net/hex
 
-    - name: Install arm-embedded toolchain
-      if: ${{ steps.builddeps-cache.outputs.cache-hit != 'true' }}
-      working-directory: /home/runner
-      run: |
-        set -euo pipefail
-        cd /home/runner
-        wget https://developer.arm.com/-/media/Files/downloads/gnu/11.3.rel1/binrel/arm-gnu-toolchain-11.3.rel1-x86_64-arm-none-eabi.tar.xz \
-        --output-document=$RUNNER_TEMP/arm-gnu-toolchain-11.3.rel1-x86_64-arm-none-eabi.tar.xz
-        tar xJf $RUNNER_TEMP/arm-gnu-toolchain-11.3.rel1-x86_64-arm-none-eabi.tar.xz
-        pwd && ls
-
     - name: Checkout and build libopencm3
       if: ${{ steps.builddeps-cache.outputs.cache-hit != 'true' }}
       working-directory: /home/runner
@@ -71,7 +60,7 @@ jobs:
       run: sudo apt update
 
     - name: "Install deps"
-      run: sudo apt install -y cmake gperf
+      run: sudo apt install -y cmake gperf gcc-arm-none-eabi
 
     - name: Checkout repo
       uses: actions/checkout@v4

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -27,6 +27,7 @@ certain VM instructions are used.
 - Fixed an issue where a timeout would occur immediately in a race condition
 - Fixed SPI close command
 - Added missing lock on socket structure
+- Fixed compilation with latest debian gcc-arm-none-eabi
 
 ## [0.6.5] - 2024-10-15
 

--- a/src/libAtomVM/term.h
+++ b/src/libAtomVM/term.h
@@ -28,6 +28,9 @@
 #ifndef _TERM_H_
 #define _TERM_H_
 
+// gcc-arm-none-eabi 13.2.1 with newlib requires this first
+#include <sys/types.h>
+
 #include <stdbool.h>
 #include <stdint.h>
 #include <stdio.h>

